stratum/hal/lib/common/common.proto

```diff
@@ -658,8 +658,35 @@ message GoogleConfig {
   map<uint64, BcmRtag7HashConfig> node_id_to_rtag7_hash_config = 7;
 }
 
+// Config specific to Tofino chassis.
+message TofinoConfig {
+  message BfPortShapingConfig {
+    message BfPerPortShapingConfig {
+      message PacketShape {
+        uint64 max_rate_pps = 1;
+        uint32 max_burst_packets = 2;
+      }
+      message ByteShape {
+        uint64 max_rate_bps = 1;
+        uint32 max_burst_bytes = 2;
+      }
+      oneof shaping {
+        PacketShape packet_shaping = 1;
+        ByteShape byte_shaping = 2;
+      }
+    }
+    // Map from port id to its shaping config given by BfPerPortShapingConfig.
+    map<uint32, BfPerPortShapingConfig> per_port_shaping_configs = 1;
```

Comment:
Just making sure, is port ID here the SDN port ID or the SDK one? From the code, it seems the former, which is good. It would be good to add a comprehensive example in the Barefoot README that shows both how to use `singleton_ports` and `vendor_config` in the same file.

Reply:
SDN port ID.

+1 on the example